diff --git a/index.js b/index.js
index 26df7859ff8d1adc50360f6d532a87c699444d67..88331b78cc462b01079bea5886af1db56c572fe6 100644
--- a/index.js
+++ b/index.js
@@ -2,8 +2,6 @@
 
 'use strict';
 
-var ascii2mathml = null;
-require('./lib/polyfills');
 
 
 function scanDelims(state, start, delimLength) {
@@ -173,28 +171,12 @@ function makeMath_block(open, close) {
 }
 
 function makeMathRenderer(renderingOptions) {
-  if (ascii2mathml === null) {
-    try {
-      ascii2mathml = require('ascii2mathml').default;
-    } catch (e) {
-      return renderingOptions && renderingOptions.display === 'block' ?
-        function(tokens, idx) {
-          return '<div class="math block">' + tokens[idx].content + '</div>';
-        } :
-        function(tokens, idx) {
-          return '<span class="math inline">' + tokens[idx].content + '</span>';
-        };
-    }
-  }
-
-  var mathml = ascii2mathml(Object.assign({}, renderingOptions));
-
   return renderingOptions && renderingOptions.display === 'block' ?
     function(tokens, idx) {
-      return mathml(tokens[idx].content) + '\n';
+      return '<div class="math block">' + tokens[idx].content + '</div>';
     } :
     function(tokens, idx) {
-      return mathml(tokens[idx].content);
+      return '<span class="math inline">' + tokens[idx].content + '</span>';
     };
 }
 
@@ -225,6 +207,22 @@ module.exports = function math_plugin(md, options) {
   md.block.ruler.after('blockquote', 'math_block', math_block, {
     alt: [ 'paragraph', 'reference', 'blockquote', 'list' ]
   });
+  md.core.ruler.push('math_inline', state => {
+    for (const token of state.tokens) {
+      if (token.type === 'html_block' && options.inlineRenderer) {
+        let result = '', lastPos = 0;
+        for (; ; ) {
+          const pOpen = token.content.indexOf(inlineOpen, lastPos);
+          if (!~pOpen) break;
+          const pClose = token.content.indexOf(inlineClose, pOpen + inlineOpen.length);
+          if (!~pClose) break;
+          result += token.content.substring(lastPos, pOpen) + options.inlineRenderer(token.content.substring(pOpen + inlineOpen.length, pClose));
+          lastPos = pClose + inlineClose.length;
+        }
+        token.content = result + token.content.substring(lastPos);
+      }
+    }
+  });
   md.renderer.rules.math_inline = inlineRenderer;
   md.renderer.rules.math_block = blockRenderer;
 };
