diff --git a/dist/index.js b/dist/index.js
index d94fd28b855b6e673f0a509ae2eb6913ac2ba59a..b0f307b59920fe709a721f6874d70fbee9ec3620 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -99,13 +99,16 @@ var Transformer = class {
       sourceFileName: filepath,
       sourceMaps: useSourceMap
     });
-    this.matches.set(filepath, matches);
+    this.matches.set(filepath, matches.map(({ name, start, end }) => {
+      const start1 = result.code.indexOf(`const ${name} = null`) + name.length + 9;
+      return { name, start, end, start1, end1: start1 + 4 };
+    }));
     return {
       code: result.code,
       map: result.map
     };
   }
-  async replaceWithData(filepath, { useSourceMap }) {
+  async replaceWithData(filepath, content_, { useSourceMap }) {
     if (!extensionsRe.test(filepath) || isInCompileScript(filepath)) {
       return;
     }
@@ -122,9 +125,8 @@ var Transformer = class {
       filepath,
       readFile(_filepath) {
         _filepath = _filepath.replace(/\\/g, "/");
-        const content2 = fs.readFileSync(_filepath, "utf-8");
         if (_filepath === filepath) {
-          const s2 = new MagicString(content2);
+          const s2 = new MagicString(content);
           s2.prepend(
             `export const ${exportName} = {};${COMPILE_TIME_FUNCTION_INJECT}`
           );
@@ -135,22 +137,23 @@ var Transformer = class {
             );
           }
           return s2.toString();
+        } else {
+          return fs.readFileSync(_filepath, "utf-8");
         }
-        return content2;
       }
     });
     const values = mod[exportName];
-    const s = new MagicString(content);
+    const s = new MagicString(content_);
     for (const m of matches) {
-      const start = m.start;
-      const end = m.end;
+      const start = m.start1;
+      const end = m.end1;
       const value = await stringifyValue(values[m.name]);
       s.overwrite(start, end, value);
     }
     if (useSourceMap) {
       return {
         code: s.toString(),
-        map: s.generateMap({ hires: true })
+        map: s.generateMap({ hires: true, includeContent: true })
       };
     }
     return {
@@ -264,8 +267,8 @@ var createPlugins = () => {
     {
       name: "compile-time:run-code",
       enforce: "pre",
-      async transform(_, id) {
-        const result = await transformer.replaceWithData(id, { useSourceMap });
+      async transform(code, id) {
+        const result = await transformer.replaceWithData(id, code, { useSourceMap });
         if (!result) return;
         if (result.dependencies) {
           result.dependencies.forEach((filepath) => {
